src/UID2.Client/SharingClient.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+
+namespace UID2.Client
+{
+    public class SharingClient
+    {
+        private TokenHelper _tokenHelper;
+
+        public SharingClient(string endpoint, string authKey, string secretKey)
+        {
+            _tokenHelper = new TokenHelper(endpoint, authKey, secretKey);
+        }
+
+        public DecryptionResponse DecryptTokenIntoRawUid(string token)
+        {
+            return _tokenHelper.Decrypt(token, DateTime.UtcNow, null, ClientType.Sharing);
+        }
+
+        public EncryptionDataResponse EncryptRawUidIntoToken(string rawUid)
```

Comment:
for completeness should there be a test for this? (it's not called anywhere?)

Reply:
per slack, I'm about to commit a bunch more tests